src/components/events/EventCard.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/Card';
+
+import { Avatar, AvatarImage } from '@/components/ui/Avatar';
+import { Badge } from '@/components/ui/Badge';
+import { Link } from '@/lib/locale/navigation';
+import { cx } from '@/lib/utils';
+import { getTranslations } from 'next-intl/server';
+
+type EventCardProps = {
+  id: number;
+  title: string;
+  subheader: string;
+  description: string;
+  imagePath?: string;
+  startTime: Date;
+  endTime: Date;
+  internal: boolean;
+  locale: string;
+  wrapperClassName?: string;
+  cardClassName?: string;
+  _active?: boolean;
+};
+
+/**
+ * A card for an event.
+ * Only set the _active prop to true if you're testing active events.
+ */
+async function EventCard(props: EventCardProps) {
+  const t = await getTranslations('events');
+  const tUi = await getTranslations('ui');
+
+  const dateOptions = {
+    hour: '2-digit',
+    minute: '2-digit',
+    hour12: false,
+  } as const;
+
+  // For example "18:00", same for end time
+  const formattedStartTime = props.startTime.toLocaleTimeString(
+    props.locale,
+    dateOptions,
+  );
+  // For example "22/8/2024" or "22.8.2024", same for end date
+  const formattedStartDate = props.startTime.toLocaleDateString(props.locale);
+  const formattedEndTime = props.endTime.toLocaleTimeString(
+    props.locale,
+    dateOptions,
+  );
+  const formattedEndDate = props.endTime.toLocaleDateString(props.locale);
+
+  const started = props.startTime < new Date() || props._active;
+  const ended = props.endTime < new Date();
+
+  return (
+    <Link
+      href={`/events/${props.id}`}
```

Comment:
when linking using a prop like id it has to be structured differently where you define the href as an object. See the next-intl documentation